core/sql/executor/ExHbaseAccess.cpp

```diff
@@ -1153,6 +1153,32 @@ Lng32 ExHbaseAccessTcb::createSQRowFromHbaseFormat(Int64 *latestRowTimestamp)
              ex_assert(FALSE, "Attr not found -2");
 	  
 	  char * defVal = attr->getDefaultValue();
+          if (! defVal)
+            {
+              char * colVal = (char*)
+                hbaseAccessTdb().listOfFetchedColNames()->get(idx);
+
+              Text colFam, colName;
+              extractColFamilyAndName(colVal, colFam, colName);
+
+              Int64 v = 0;
+              if (colName.length() == sizeof(char))
+                v = *(char*)colName.data();
+              else if (colName.length() == sizeof(unsigned short))
+                v = *(UInt16*)colName.data();
+              else if (colName.length() == sizeof(Lng32))
+                v = *(ULng32*)colName.data();
+
+              char buf[10];
```

Comment:
2^30 = 1073741824, which is one byte too big to fit in buf (since you need room for a null terminator). We are unlikely to see that many columns, but still would be nice to make buf 11 bytes at least.